route/build/src/pio_utils.f90

```diff
@@ -692,19 +692,18 @@ SUBROUTINE write_char0D(pioFileDesc,     &
   ierr=0; message='write_char0D/'
 
   if ( len(string) > size(tmpString) )then
-      ierr = 1
-      message=trim(message)//'ERROR: length of string being written is larger than tmpString'
-      return
+    ierr = 1
+    message=trim(message)//'ERROR: length of string being written is larger than tmpString'
+    return
   end if
   ierr = pio_inq_varid(pioFileDesc, trim(vname), pioVarId)
   if(ierr/=0)then; message=trim(message)//'ERROR: getting variable id'; return; endif
 
-
   do m = 1,len(string)
-     tmpString(m:m) = string(m:m)
+    tmpString(m:m) = string(m:m)
   end do
-  start(1) = iStart
-  start(2) = 1
+  start(1) = 1
```

Comment:
I think you are right here. The first dimension is the number of characters and the second is the number of history files.